sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/CustomizedHeadersTest.java

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.data.appconfiguration;
+
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.HttpMethod;
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.HttpPipelineBuilder;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.FixedDelay;
+import com.azure.core.http.policy.RetryPolicy;
+import com.azure.core.test.http.NoOpHttpClient;
+import com.azure.data.appconfiguration.implementation.CustomHeadersPolicy;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.net.URL;
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.UUID;
+import java.util.function.Supplier;
+
+public class CustomizedHeadersTest {
+
+    private static final String REQUEST_ID_HEADER = "x-ms-client-request-id";
+    private static final String CORRELATION_ID_HEADER = "x-ms-correlation-request-id";
+    private static final String CORRELATION_CONTEXT_HEADER = "correlation-context";
+
```

Comment:
My earlier PR has exactly same solution in https://github.com/Azure/azure-sdk-for-java/pull/6400   

@mssfang  We talked about this today morning,  Since you need these three Header and you did not feel comfortable putting these headers in MessageIDPolicy because they are not all related to MessageID.

Reply:
Is that possible to add the Supplier to the AddHeaderPolicy. It seems odd to me by supporting other customized headers in the RequestIDPolicy